core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/internal/DefaultMetricCollection.java

```diff
@@ -0,0 +1,68 @@
+package software.amazon.awssdk.metrics.internal;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import software.amazon.awssdk.metrics.MetricCollection;
+import software.amazon.awssdk.metrics.MetricRecord;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+public class DefaultMetricCollection implements MetricCollection {
+    private final String name;
+    private final Map<SdkMetric<?>, List<MetricRecord<?>>> metrics;
+    private final Collection<MetricCollection> children;
+
+
+    public DefaultMetricCollection(String name, Map<SdkMetric<?>, List<MetricRecord<?>>> metrics, Collection<MetricCollection> children) {
+        this.name = name;
+        this.metrics = metrics;
+        this.children = children != null ? Collections.unmodifiableCollection(children) : Collections.emptyList();
+    }
```

Comment:
We should be clear in the Javadoc that we're not doing copies of these inputs, since in other places we do that.